library/src/main/java/com/novoda/downloadmanager/lib/BatchQuery.java

```diff
@@ -211,29 +221,45 @@ public Builder withExtraData(String extraData) {
         }
 
         public BatchQuery build() {
-            if (criteriaIdBuilder != null) {
-                builder.withInnerCriteria(criteriaIdBuilder.build());
-                if (criteriaStatusBuilder != null) {
+
+            List<Criteria.Builder> criteriaBuilders = setCriteriaBuilders();
+            setCriteriaListFrom(criteriaBuilders);
+
+            Criteria criteria = builder.build();
+            String selection = criteria.getSelection();
+            String sortOrder = criteria.getSort();
+            String[] selectionArguments = criteria.getSelectionArguments();
+            return new BatchQuery(selection, selectionArguments, sortOrder);
+        }
+
+        private void setCriteriaListFrom(List<Criteria.Builder> criteriaBuilders) {
```

Comment:
are there any tests around setting multiple criteria? Looks complex and would be good to make sure we don't break it :)